### ISSUE TITLE
Fresh install via composer create-project ends in exception and database error

### ISSUE THREAD

**Issue by Bubelbub** [Prophet731/BFAdminCP#224]
Hey guys,

**installation log:**

	C:\xampp\htdocs>composer create-project --prefer-dist --no-scripts --ignore-platform-reqs --keep-vcs adkgamers/bfadmincp BFAdminCP
	Installing adkgamers/bfadmincp (v2.0.3)
	  - Installing adkgamers/bfadmincp (v2.0.3)
	    Cloning 5b49a28091f87da2c4e6e753c7cafff3b966b8ba
	
	Created project in BFAdminCP
	Loading composer repositories with package information
	Installing dependencies (including require-dev) from lock file
	  - Installing symfony/translation (v2.5.12)
	    Loading from cache
	
	  - Installing symfony/security-core (v2.5.12)
	    Loading from cache
	
	  - Installing symfony/routing (v2.5.12)
	    Loading from cache
	
	  - Installing symfony/process (v2.5.12)
	    Loading from cache
	
	  - Installing symfony/http-foundation (v2.5.12)
	    Loading from cache
	
	  - Installing symfony/event-dispatcher (v2.8.1)
	    Loading from cache
	
	  - Installing symfony/debug (v2.5.12)
	    Loading from cache
	
	  - Installing psr/log (1.0.0)
	    Loading from cache
	
	  - Installing symfony/http-kernel (v2.5.12)
	    Loading from cache
	
	  - Installing symfony/finder (v2.5.12)
	    Loading from cache
	
	  - Installing symfony/dom-crawler (v2.5.12)
	    Loading from cache
	
	  - Installing symfony/css-selector (v2.5.12)
	    Loading from cache
	
	  - Installing symfony/console (v2.5.12)
	    Loading from cache
	
	  - Installing symfony/browser-kit (v2.5.12)
	    Loading from cache
	
	  - Installing swiftmailer/swiftmailer (v5.4.1)
	    Loading from cache
	
	  - Installing stack/builder (v1.0.3)
	    Loading from cache
	
	  - Installing predis/predis (v0.8.7)
	    Loading from cache
	
	  - Installing phpseclib/phpseclib (0.3.10)
	    Loading from cache
	
	  - Installing patchwork/utf8 (v1.3.0)
	    Loading from cache
	
	  - Installing nesbot/carbon (1.17.0)
	    Loading from cache
	
	  - Installing monolog/monolog (1.17.2)
	    Loading from cache
	
	  - Installing nikic/php-parser (v0.9.5)
	    Loading from cache
	
	  - Installing jeremeamia/superclosure (1.0.2)
	    Loading from cache
	
	  - Installing ircmaxell/password-compat (v1.0.4)
	    Loading from cache
	
	  - Installing filp/whoops (1.1.10)
	    Loading from cache
	
	  - Installing d11wtq/boris (v1.0.10)
	    Loading from cache
	
	  - Installing symfony/filesystem (v2.8.1)
	    Loading from cache
	
	  - Installing classpreloader/classpreloader (1.0.2)
	    Loading from cache
	
	  - Installing laravel/framework (v4.2.17)
	    Loading from cache
	
	  - Installing anahkiasen/html-object (1.4.2)
	    Loading from cache
	
	  - Installing anahkiasen/former (3.5.9)
	    Loading from cache
	
	  - Installing pusher/pusher-php-server (v2.2.1)
	    Loading from cache
	
	  - Installing artdarek/pusherer (1.0.2)
	    Loading from cache
	
	  - Installing tedivm/jshrink (v1.1.0)
	    Loading from cache
	
	  - Installing natxet/cssmin (v3.0.4)
	    Loading from cache
	
	  - Installing ceesvanegmond/minify (2.0.13)
	    Loading from cache
	
	  - Installing davejamesmiller/laravel-breadcrumbs (2.3.1)
	    Loading from cache
	
	  - Installing league/fractal (0.11.0)
	    Loading from cache
	
	  - Installing dingo/api (v0.8.2)
	    Loading from cache
	
	  - Installing maxmind/web-service-common (v0.0.4)
	    Loading from cache
	
	  - Installing maxmind-db/reader (v1.0.3)
	    Loading from cache
	
	  - Installing geoip2/geoip2 (v2.3.3)
	    Loading from cache
	
	  - Installing react/promise (v2.2.1)
	    Loading from cache
	
	  - Installing guzzlehttp/streams (3.0.0)
	    Loading from cache
	
	  - Installing guzzlehttp/ringphp (1.1.0)
	    Loading from cache
	
	  - Installing guzzlehttp/guzzle (5.3.0)
	    Loading from cache
	
	  - Installing kmd/logviewer (1.2.2)
	    Loading from cache
	
	  - Installing lavary/laravel-menu (v1.5.0)
	    Loading from cache
	
	  - Installing paragonie/random_compat (1.1.4)
	    Loading from cache
	
	  - Installing roumen/feed (v2.8.7)
	    Loading from cache
	
	  - Installing vierbergenlars/php-semver (3.0.0)
	    Loading from cache
	
	  - Installing zizaco/confide (dev-master a5ba244)
	    Loading from cache
	
	  - Installing laravelbook/ardent (v2.5.0)
	    Loading from cache
	
	  - Installing zizaco/entrust (1.2.5)
	    Loading from cache
	
	symfony/translation suggests installing symfony/config ()
	symfony/translation suggests installing symfony/yaml ()
	symfony/security-core suggests installing symfony/expression-language (For using the expression voter)
	symfony/security-core suggests installing symfony/validator (For using the userpassword constraint)
	symfony/routing suggests installing doctrine/annotations (For using the annotation loader)
	symfony/routing suggests installing symfony/config (For using the all-in-one router or any loader)
	symfony/routing suggests installing symfony/expression-language (For using expression matching)
	symfony/routing suggests installing symfony/yaml (For using the YAML loader)
	symfony/event-dispatcher suggests installing symfony/dependency-injection ()
	symfony/http-kernel suggests installing symfony/class-loader ()
	symfony/http-kernel suggests installing symfony/config ()
	symfony/http-kernel suggests installing symfony/dependency-injection ()
	predis/predis suggests installing ext-phpiredis (Allows faster serialization and deserialization of the Redis protocol)
	phpseclib/phpseclib suggests installing ext-gmp (Install the GMP (GNU Multiple P
	recision) extension in order to speed up arbitrary precision integer arithmeticoperations.)
	phpseclib/phpseclib suggests installing pear-pear/PHP_Compat (Install PHP_Compat to get phpseclib working on PHP < 4.3.3.)
	patchwork/utf8 suggests installing ext-intl (Use Intl for best performance)
	patchwork/utf8 suggests installing ext-wfio (Use WFIO for UTF-8 filesystem access on Windows)
	monolog/monolog suggests installing aws/aws-sdk-php (Allow sending log messagesto AWS services like DynamoDB)
	monolog/monolog suggests installing doctrine/couchdb (Allow sending log messages to a CouchDB server)
	monolog/monolog suggests installing ext-amqp (Allow sending log messages to an AMQP server (1.0+ required))
	monolog/monolog suggests installing ext-mongo (Allow sending log messages to a MongoDB server)
	monolog/monolog suggests installing graylog2/gelf-php (Allow sending log messages to a GrayLog2 server)
	monolog/monolog suggests installing php-console/php-console (Allow sending log messages to Google Chrome)
	monolog/monolog suggests installing raven/raven (Allow sending log messages to a Sentry server)
	monolog/monolog suggests installing rollbar/rollbar (Allow sending log messagesto Rollbar)
	monolog/monolog suggests installing ruflin/elastica (Allow sending log messagesto an Elastic Search server)
	monolog/monolog suggests installing videlalvaro/php-amqplib (Allow sending log messages to an AMQP server using php-amqplib)
	laravel/framework suggests installing doctrine/dbal (Allow renaming columns anddropping SQLite columns.)
	league/fractal suggests installing pagerfanta/pagerfanta (Pagerfanta Paginator)
	league/fractal suggests installing zendframework/zend-paginator (Zend FrameworkPaginator)
	dingo/api suggests installing lucadegasperi/oauth2-server-laravel (Use the League OAuth 2.0 server to protect your API.)
	dingo/api suggests installing tymon/jwt-auth (Protect your API with JSON Web Tokens)
	paragonie/random_compat suggests installing ext-libsodium (Provides a modern crypto API that can be used to generate random bytes.)
	Generating autoload files

**First start of the application.**

`Error in exception handler: Class bfadmincp does not exist in C:\xampp\htdocs\BFAdminCP\vendor\laravel\framework\src\Illuminate\Container\Container.php:504`

**Database after first start.**

![mysql](https://cloud.githubusercontent.com/assets/319352/16814126/b5a6217c-4934-11e6-9c8e-9a9506566ba2.png)

**After second start.**

	Database Error!
	
	Sorry, something went wrong with the database.

What can I do? :+1:

**Comment by killengage** [comment]
post the /storage/log/ log file. The file not copy paste the content.

**Comment by Prophet731** [comment]
Delete all tables that start with bfacp_ then cd to the project die and run php artisan migrate - - force next run php artisan db:seed - - force does that fix the problem?

**Comment by Prophet731** [comment]
Dir*

**Comment by Bubelbub** [comment]
@Prophet731: Thank you for help but...

	www-data@webserver:~/bfadmincp$ php artisan migrate
	**************************************
	*     Application In Production!     *
	**************************************
	
	Do you really wish to run this command? y
	Migration table created successfully.
	SQLSTATE[HY000]: General error: 1005 Can't create table 'bfadmincp.#sql-41e_79de' (errno: 150) (SQL: alter table `bfacp_users_soldiers` add constraint bfacp_users_soldiers_player_id_foreign foreign key (`player_id`) references `tbl_playerdata` (`PlayerID`) on delete cascade on update cascade)
	www-data@webserver:~/bfadmincp$ php artisan migrate
	**************************************
	*     Application In Production!     *
	**************************************
	
	Do you really wish to run this command? y
	SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'bfacp_options' already exists (SQL: create table `bfacp_options` (`option_id` int unsigned not null auto_increment primary key, `option_key` varchar(64) not null, `option_title` varchar(100) not null, `option_value` longtext null, `option_description` varchar(255) null) default character set utf8 collate utf8_unicode_ci engine = InnoDB)
	www-data@webserver:~/bfadmincp$ php artisan migrate
	**************************************
	*     Application In Production!     *
	**************************************
	
	Do you really wish to run this command? y
	Migration table created successfully.
	SQLSTATE[HY000]: General error: 1005 Can't create table 'bfadmincp.#sql-41e_79e0' (errno: 150) (SQL: alter table `bfacp_users_soldiers` add constraint bfacp_users_soldiers_player_id_foreign foreign key (`player_id`) references `tbl_playerdata` (`PlayerID`) on delete cascade on update cascade)
	www-data@webserver:~/bfadmincp$ php artisan db:seed - - force
	Too many arguments.
	www-data@webserver:~/bfadmincp$ php artisan db:seed
	**************************************
	*     Application In Production!     *
	**************************************
	
	Do you really wish to run this command? y
	Seeded: OptionsTableSeeder
	SQLSTATE[42S02]: Base table or view not found: 1146 Table 'bfadmincp.bfacp_roles' doesn't exist (SQL: select count(*) as aggregate from `bfacp_roles` where `name` = Administrator)

I tried to delete all tables after each step.

@killengage:

    [2016-07-14 21:21:04] production.ERROR: exception 'PDOException' with message 'SQLSTATE[HY000]: General error: 1005 Can't create table 'bfadmincp.#sql-41e_79d1' (errno: 150)' in /var/www/bfadmincp/vendor/laravel/framework/src/Illuminate/Database/Connection.php:369
    Next exception 'Illuminate\Database\QueryException' with message 'SQLSTATE[HY000]: General error: 1005 Can't create table 'bfadmincp.#sql-41e_79d1' (errno: 150) (SQL: alter table `bfacp_users_soldiers` add constraint bfacp_users_soldiers_player_id_foreign foreign key (`player_id`) references `tbl_playerdata` (`PlayerID`) on delete cascade on update cascade)' in /var/www/bfadmincp/vendor/laravel/framework/src/Illuminate/Database/Connection.php:625

**Comment by Bubelbub** [comment]
Ah @Prophet731: maybe there is something wrong configured? Or must I install something else before installing this control panel?

In this file https://github.com/Prophet731/BFAdminCP/blob/master/app/bfacp/Http/Controllers/Api/HelpersController.php you mention a table `table('tbl_currentplayers')` or `adkats_usersoldiers`.

I think the tables are missing because I not installed this plugins...???

**Comment by killengage** [comment]
You have to install "Adkats" plugin and the "Chat, GUID, Stats and Mapstats Logger" plugin for Procon.
After installing these delete all BCFAP tables.

**Comment by Bubelbub** [comment]
Ok, @killengage.
Now I see something :+1: 

And in administration:
![2016-07-15 00_17_28-mremoteng - confcons xml](https://cloud.githubusercontent.com/assets/319352/16857660/9ebe4aa0-4a21-11e6-8295-3a9cc585e916.png)

**Comment by Prophet731** [comment]
Again drop the tables `bfacp_*` then run the commands again, don't delete the tables after each command, that would defeat the purpose.

**Comment by Bubelbub** [comment]
I get this error 500 again.
Where can I download the 2.1? Thanks!

**Comment by Prophet731** [comment]
Did you also run the db:seed command as well or just the migrate one?

**Comment by Bubelbub** [comment]
Yes

**Comment by Bubelbub** [comment]
@Prophet731 @killengage how to download 2.1?
Your 2.0.3 tag equals the master branch.

**Comment by killengage** [comment]
You have to switch to the "dev-laravel5" branch on git. I advise you to go for the composer/git method.

**Comment by Prophet731** [comment]
Sorry was on vacation this weekend. To install the dev copy see this https://github.com/Prophet731/BFAdminCP/tree/dev-laravel5#installing-dev-version

**Comment by Bubelbub** [comment]
Thank you very much.
But it work sinnce yesterday.

I ignored the composer create-project method and used git clone instead
with composer install. (Dev-Mode)
Then I had same issues with PHP7 - but I was able to deal with this by
updating some composer.json packages.
And then I changed path for CURL HTTPS ca-certificates in php.ini and the
update function works, too.

:-)

2016-07-18 23:00 GMT+02:00 Prophet731 <notifications@github.com>:
